spec/custom-provider-spec.js

```diff
@@ -3,23 +3,48 @@
 import fs from 'fs-extra';
 import temp from 'temp';
 import CustomFile from '../lib/atom-build.js';
+import os from 'os';
 
 describe('custom provider', () => {
   let builder;
+  let createdHomeDir;
   let directory = null;
+  const originalHomedirFn = os.homedir;
 
   temp.track();
 
   beforeEach(() => {
+    createdHomeDir = temp.mkdirSync('atom-build-spec-home');
     directory = fs.realpathSync(temp.mkdirSync({ prefix: 'atom-build-spec-' })) + '/';
     atom.project.setPaths([ directory ]);
+    os.homedir = () => createdHomeDir;
     builder = new CustomFile(directory);
   });
 
   afterEach(() => {
+    os.homedir = originalHomedirFn;
     fs.removeSync(directory);
   });
 
+  it('when there is no .atom-build config file in any elegible directory', () => {
+    expect(builder.isEligible()).toEqual(false);
+  });
+
+  describe('when .atom-build config is on home directory', () => {
+    it('should find json file in home directory', () => {
+      fs.writeFileSync(createdHomeDir + '/.atom-build.json', fs.readFileSync(__dirname + '/fixture/.atom-build.json'));
+      expect(builder.isEligible()).toEqual(true);
+    });
+    it('should find cson file in home directory', () => {
+      fs.writeFileSync(createdHomeDir + '/.atom-build.cson', fs.readFileSync(__dirname + '/fixture/.atom-build.cson'));
+      expect(builder.isEligible()).toEqual(true);
+    });
+    it('should find yml file in home directory', () => {
+      fs.writeFileSync(createdHomeDir + '/.atom-build.yml', fs.readFileSync(__dirname + '/fixture/.atom-build.yml'));
+      expect(builder.isEligible()).toEqual(true);
+    });
+  });
```

Comment:
@noseglid I rethought this and I think it is not necessary, since this tests are about being able to read the file from home directory and it does it, It is not about parsing targets.

That being said, I think the tests after mine it handles the parsing and checking the files are being parsed.

I would suggest we better split the tests logic into: 1, check it is able to find a config file from the directory/project, 2: check it is able to parse the config file and the parsed data is correct.

Otherwise we would be me mixing two differents test into one.

I updated the PR with the async logic for home directory, realpath checking and the extra line in the os import.

Please advise and looking forward to keep contributing if you want!